app/models/comment.rb

```diff
@@ -12,6 +12,8 @@ class Comment < ApplicationRecord
   TITLE_DELETED = "[deleted]".freeze
   TITLE_HIDDEN = "[hidden by post author]".freeze
 
+  URI_REGEXP = %r{(?<scheme>https?://)?(?<host>.+?)(?<port>:\d+)?$}.freeze
```

Comment:
Thanks for the suggestions. The current regex is not the best, it's just something that works, I'll definitely be trying those out. I left in the captures for readability purposes, they aren't needed otherwise.

Reply:
I found the implementation with the `x` modifier more readable, it prevents nils as well since there's only one capture group. I have changed the regex to reflect this.